docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc

```diff
@@ -1244,6 +1244,81 @@ If you set `quarkus.oidc.client-id`, but your endpoint does not require remote a
 Quarkus `web-app` applications always require the `quarkus.oidc.client-id` property.
 ====
 
+== Mutual TLS token binding
+
+https://datatracker.ietf.org/doc/html/rfc8705[RFC8705] describes a mechanism for binding access tokens to Mutual TLS (mTLS) client authentication certificates.
+It requires that a client certificate's SHA256 thumbprint matches a JWT token or token introspection confirmation `x5t#S256` certificate thumbprint.
+
+For example, see https://datatracker.ietf.org/doc/html/rfc8705#section-3.1[JWT Certificate Thumbprint Confirmation Method] and https://datatracker.ietf.org/doc/html/rfc8705#section-3.2[Confirmation Method for Token Introspection] sections of https://datatracker.ietf.org/doc/html/rfc8705[RFC8705].
+
+MTLS token binding supports a `holder of key` concept, and can be used to confirm that the current access token was issued to the current authenticated client who presents this token.
+
+When you use both mTLS and OIDC bearer authentication mechanisms, you can enforce that the access tokens must be certificate bound with a single property, after configuring your Quarkus endpoint and Quarkus OIDC to require the use of mTLS.
+
+For example:
+
+[source,properties]
+----
+quarkus.oidc.auth-server-url=${your_oidc_provider_url}
+quarkus.oidc.token.certificate-bound=true <1>
```

Comment:
Would it make sense to have a specific namespace for sender-constrained tokens such as `quarkus.oidc.token.binding` and have certificate-bound tokens configured as `quarkus.oidc.token.binding.certificate=true`?

Thinking about DPoP, for instance, we could use the same namespace to configure it. Something like `quarkus.oidc.token.binding.dpop.<properties>`.

It could also help to add more capabilities, when needed, for each individual token-binding method.

Not sure, only a suggestion.

Reply:
Hi @pedroigor I think you are right, thanks for thinking ahead, we'd want to have dpop configured at the same level. I'll work on encapsulating this configuration as you suggested

Reply:
Hey @pedroigor, I've encapsulated it as `token.binding.certificate`. So far it is the only properties, but one case I've heard about is about TLS termination so we may want to support a header passing the cert thumbprint internally for those cases, and like you said, other binding customizations will be easier to handle.

I'm not sure yet if we will put `dpop` inside or next to `binding` as it is a bit different, and there is also dpop related to the authorization code flow, but we can discuss it in a month or two

thanks